### ISSUE TITLE
Help in the installation of MSM

### ISSUE THREAD

**Issue by marivas-MRI** [ecr05/MSM_HOCR_macOSX#24]
Dear all, 
I am having difficulty installing MSM on my ubuntu 18.04. I have tried to follow the installation steps detailed in the latest MSM release. 

1. Create a folder named MSM_software where I created a folder named extras (with subfolders include and src)

Then I added this line in my bashrc file   export FSLDEVDIR=export FSLDEVDIR=/home/miguel/MSM_software/

2. Copy the ELC folder in /home/miguel/MSM_software/extras/include/ERC

3. Copy the /usr/local/fsl/src/FastPDlib to /home/miguel/MSM_software/extras/src/FastPDlib

Now I should compile each repository directory using the male install command. However, should I use this command to compile each directory. For example, I open a terminal in the /usr/local/fsl/include/newmesh folder and I get this error when I type make install

make: *** No rule to make target 'install'.  Stop.

Please, can anyone tell me how to compile each directory to finally get the MSM binaries?.

Thank you very much.

**Comment by ecr05** [comment]
Do you need to actually install it or can you not use the ubuntu binary from the release? FSL doesn't compile for ubuntu 18 so if you are building everything from scratch you are going to get stuck

**Comment by marivas-MRI** [comment]
I am trying to install MSM to use it in the HCP preprocessing pipeline.

**Comment by ecr05** [comment]
Yes well you don't need to as there are precompiled binaries. See Readme where there is a link

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by marivas-MRI** [comment]
Sorry but I am trying to find the link in the Readme file but I can't find it. In the Readme file I only found this link

https://github.com/ecr05/MSM_HOCR/blob/master/compilation%20instructions

Would you be so kind as to send me the link you mentioned where I can download the precompiled binaries?. Sorry for such a basic question and thank you for your patience!

**Comment by srajp-del** [comment]
> Dear all, I am having difficulty installing MSM on my ubuntu 18.04. I have tried to follow the installation steps detailed in the latest MSM release.
> 
>     1. Create a folder named MSM_software where I created a folder named extras (with subfolders include and src)
> 
> 
> Then I added this line in my bashrc file export FSLDEVDIR=export FSLDEVDIR=/home/miguel/MSM_software/
> 
>     2. Copy the ELC folder in /home/miguel/MSM_software/extras/include/ERC
> 
>     3. Copy the /usr/local/fsl/src/FastPDlib to /home/miguel/MSM_software/extras/src/FastPDlib
> 
> 
> Now I should compile each repository directory using the male install command. However, should I use this command to compile each directory. For example, I open a terminal in the /usr/local/fsl/include/newmesh folder and I get this error when I type make install
> 
> make: *** No rule to make target 'install'. Stop.
> 
> Please, can anyone tell me how to compile each directory to finally get the MSM binaries?.
> 
> Thank you very much.

hi,
I think if you do these 2 step before using make your problem will be solved :
export FSLDEVDIR=/usr/local/fsl
export FSLCONFDIR=$FSLDIR/config

**Comment by ecr05** [comment]
You should be able to download a binary that works for you from  here

https://github.com/ecr05/MSM_HOCR/releases



Dr Emma C. Robinson

https://metrics-lab.github.io/


Apologies if you get emails outside normal work hours - I have small children and work flexibly. I don't expect a reply at these times.


Senior Lecturer In Medical Image Computing

School of Biomedical Engineering and Imaging Sciences

Kings College London


Office: 5th floor, Becket House, 1 Lambeth Palace Road, London SE1 7EU, UK
Mail: 4th Floor, Lambeth Wing, St Thomas' Hospital, London SE1 7EH
Tel. +44 (0)20 784 89557

________________________________

**Comment by srajp-del** [comment]
> You should be able to download a binary that works for you from here https://github.com/ecr05/MSM_HOCR/releases Dr Emma C. Robinson https://metrics-lab.github.io/ Apologies if you get emails outside normal work hours - I have small children and work flexibly. I don't expect a reply at these times. Senior Lecturer In Medical Image Computing School of Biomedical Engineering and Imaging Sciences Kings College London Office: 5th floor, Becket House, 1 Lambeth Palace Road, London SE1 7EU, UK Mail: 4th Floor, Lambeth Wing, St Thomas' Hospital, London SE1 7EH Tel. +44 (0)20 784 89557
> […](#)
> ________________________________ From: Shirin353 ***@***.***> Sent: Wednesday, January 4, 2023 12:29 PM To: ecr05/MSM_HOCR ***@***.***> Cc: Emma Robinson ***@***.***>; Comment ***@***.***> Subject: Re: [ecr05/MSM_HOCR] Help in the installation of MSM (Issue #24) Dear all, I am having difficulty installing MSM on my ubuntu 18.04. I have tried to follow the installation steps detailed in the latest MSM release. 1. Create a folder named MSM_software where I created a folder named extras (with subfolders include and src) Then I added this line in my bashrc file export FSLDEVDIR=export FSLDEVDIR=/home/miguel/MSM_software/ 2. Copy the ELC folder in /home/miguel/MSM_software/extras/include/ERC 3. Copy the /usr/local/fsl/src/FastPDlib to /home/miguel/MSM_software/extras/src/FastPDlib Now I should compile each repository directory using the male install command. However, should I use this command to compile each directory. For example, I open a terminal in the /usr/local/fsl/include/newmesh folder and I get this error when I type make install make: *** No rule to make target 'install'. Stop. Please, can anyone tell me how to compile each directory to finally get the MSM binaries?. Thank you very much. hi, I think if you do these 2 step before using make your problem will be solved : export FSLDEVDIR=/usr/local/fsl export FSLCONFDIR=$FSLDIR/config — Reply to this email directly, view it on GitHub<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fecr05%2FMSM_HOCR%2Fissues%2F24%23issuecomment-1370869105&data=05%7C01%7Cemma.robinson%40kcl.ac.uk%7C81c1ed99ea1348352b9808daee4f4d5e%7C8370cf1416f34c16b83c724071654356%7C0%7C0%7C638084321610612872%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=cKtR%2Fj8Cs2f5yZaBTNOCfXuZ1VlbIiAm7bXWzQjyLEc%3D&reserved=0>, or unsubscribe<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAD7GXPACDAA6ZTFOVVPC2A3WQVUJ5ANCNFSM6AAAAAASCFSCNI&data=05%7C01%7Cemma.robinson%40kcl.ac.uk%7C81c1ed99ea1348352b9808daee4f4d5e%7C8370cf1416f34c16b83c724071654356%7C0%7C0%7C638084321610769110%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=jGFv%2Bsa%2BWWA6yFcGuL2XANsPHTD5pCxi0KQt4p4xP%2B4%3D&reserved=0>. You are receiving this because you commented.Message ID: ***@***.***>

Hi,
Thanks for your help,I run the bash file and it gets ok.
